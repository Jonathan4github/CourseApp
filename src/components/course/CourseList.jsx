import React from 'react';
import PropTypes from 'prop-types';
import CourseListRow from './CourseListRow';


const propTypes = {
  courses: PropTypes.array.isRequired,
  handleDelete: PropTypes.func.isRequired
};

const CourseList = ({ courses, handleDelete }) => (
  <table className="table table-hover">
    <thead>
      <tr>
        <th>&nbsp;</th>
        <th>Title</th>
        <th>Author</th>
        <th>Category</th>
        <th>Length</th>
        <th>Actions</th>
      </tr>
    </thead>
    <tbody>
      {courses
        .map(course => (
          <CourseListRow
            key={course.id}
            course={course}
            handleDelete={handleDelete}
            />
        ))
      }
    </tbody>
  </table>
);

CourseList.propTypes = propTypes;

export default CourseList;
