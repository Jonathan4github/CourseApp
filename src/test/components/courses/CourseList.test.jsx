/* eslint-disable */
import CourseList from '../../../components/course/CourseList';
import courseMock from '../../__mocks__/courseMock';


describe('<CourseList /> component', () => {
  const setup = () => {
    const props = {
      courses: courseMock.courses,
      handleDelete: jest.fn()
    };

    return {
      shallowWrapper: shallow(<CourseList {...props} />),
      props
    };
  };


  it('should render without crashing', () => {
    const { shallowWrapper } = setup();

    expect(toJson(shallowWrapper)).toMatchSnapshot();
  });

  it('should check that proper table element are rendered', () => {
    const { shallowWrapper, props } = setup();

    expect(shallowWrapper.find('table').children()).toHaveLength(2);
    expect(shallowWrapper.find('tr').children()).toHaveLength(6);
    expect(shallowWrapper.find('CourseListRow')).toHaveLength(props.courses.length);
  });
});
