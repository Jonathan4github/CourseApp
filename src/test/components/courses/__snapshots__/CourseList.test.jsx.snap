// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CourseList /> component should render without crashing 1`] = `
<table
  className="table table-hover"
>
  <thead>
    <tr>
      <th>
        Link
      </th>
      <th>
        Title
      </th>
      <th>
        Author
      </th>
      <th>
        Category
      </th>
      <th>
        Length
      </th>
      <th>
        Actions
      </th>
    </tr>
  </thead>
  <tbody>
    <CourseListRow
      course={
        Object {
          "authorId": "cory-house",
          "category": "JavaScript",
          "id": "react-flux-building-applications",
          "length": "5:08",
          "title": "Building Applications in React and Flux",
          "watchHref": "http://www.pluralsight.com/courses/react-flux-building-applications",
        }
      }
      handleDelete={[MockFunction]}
      key="react-flux-building-applications"
    />
    <CourseListRow
      course={
        Object {
          "authorId": "cory-house",
          "category": "Software Practices",
          "id": "clean-code",
          "length": "3:10",
          "title": "Clean Code: Writing Code for Humans",
          "watchHref": "http://www.pluralsight.com/courses/writing-clean-code-humans",
        }
      }
      handleDelete={[MockFunction]}
      key="clean-code"
    />
  </tbody>
</table>
`;
